src/bip44.c

```diff
@@ -212,10 +247,100 @@ size_t bip44_printToStr(const bip44_path_t* pathSpec, char* out, size_t outSize)
 	return ptr - out;
 }
 
+
+bip44_path_type_t bip44_classifyPath(const bip44_path_t* pathSpec)
+{
+	if (bip44_hasValidCardanoPoolColdKeyPrefix(pathSpec)) {
+		if (bip44_isValidPoolColdKeyPath(pathSpec)) {
+			return PATH_POOL_COLD_KEY;
+		} else {
+			return PATH_INVALID;
+		}
+	}
+
+	if (bip44_hasValidCardanoWalletPrefix(pathSpec)) {
+		switch (pathSpec->length) {
+
+		case 3: {
+			return PATH_WALLET_ACCOUNT;
+		}
+
+		case 5: {
+			const uint8_t chainType = bip44_getChainTypeValue(pathSpec);
+			switch (chainType) {
+
+			case CARDANO_CHAIN_INTERNAL:
+			case CARDANO_CHAIN_EXTERNAL:
+				return PATH_WALLET_SPENDING_KEY;
+
+			case CARDANO_CHAIN_STAKING_KEY:
+				if (bip44_isValidStakingKeyPath(pathSpec)) {
+					return PATH_WALLET_STAKING_KEY;
+				} else {
+					return PATH_INVALID;
+				}
+
+			default:
+				return PATH_INVALID;
+			}
+		}
+
+		default:
+			return PATH_INVALID;
+		}
+	}
+
+	return PATH_INVALID;
```

Comment:
Is there some common coding ruleset for the company or the project I can read about? These four returns (instead of just the one at the end) might improve or worsen readability, depending on the stance of the people involved, and I'm curious if there's a written one.

Reply:
I don't think we have such a ruleset for hardware wallets or Adalite (and definitely not one company-wide, at least not one I know of; there are many projects and they are quite diverse, but I believe some bigger teams might have some rules). We use some code formatters, though (eslint for js, astyle for C).

In this case, there are many reasons why a path is invalid, and new ones are popping up every other while as new valid path schemes are added. This particular method will be refactored (split into several parts at least) soon when multisig is added, i.e. paths starting with 1854' instead of 1852' (https://github.com/cardano-foundation/CIPs/blob/master/CIP-1854/CIP-1854.md). Code improvement suggestions are always welcome, though. I will think about whether do some changes right away.